cmd/recv.go

```diff
@@ -199,6 +199,21 @@ func recvAction(cmd *cobra.Command, args []string) {
 					bail("Read zip error: %s", err)
 				}
 
+				// calculate the uncompressed size of the contents of the zip
+				var actualUncompressedSize uint64
+				var fileCount int
+				for _, f := range zr.File {
+					actualUncompressedSize += f.FileHeader.UncompressedSize64
+					fileCount++
+				}
+				if msg.UncompressedBytes64 < int64(actualUncompressedSize) {
+					bail("error: uncompressed size of the directory mismatch with that in the offer message")
+				}
+
+				if msg.FileCount < fileCount {
+					bail("error: number of files in the directory mismatch with that in the offer message")
+				}
+
```

Comment:
I realised that running `fileCount++` in the loop actually is quite unnecessary. We can just use the size of the slice instead and avoid one add instruction for each loop iteration. 
```suggestion
				// calculate the uncompressed size of the contents of the zip
				var actualUncompressedSize uint64
				for _, f := range zr.File {
					actualUncompressedSize += f.FileHeader.UncompressedSize64
				}
				if msg.UncompressedBytes64 < int64(actualUncompressedSize) {
					bail("error: uncompressed size of the directory mismatch with that in the offer message")
				}

				if msg.FileCount < len(zr.File) {
					bail("error: number of files in the directory mismatch with that in the offer message")
				}

```

Reply:
I updated the suggestion as I realised it actually makes more sense to check file counts before checking uncompressed size in this case.

Reply:
@vu3rdd Any chance to get this improved so the PR perhaps can be merged after that?